dotnet-coverage-test-data-import-guide.md

```diff
@@ -1,117 +1,137 @@
+
+    * sonar.cs.vscoveragexml.reportsPaths
+        * Path to Visual Studio Code Coverage report. Multiple paths may be comma-delimited, or included via wildcards.
+        * Ensure that you have converted the code coverage report from Binary into XML (see "Convert the Code Coverage Report from Binary into XML" section on our [[Coverage & Test Data] Generate Reports for C#, VB.net](https://community.sonarsource.com/t/coverage-test-data-generate-reports-for-c-vb-net/9871) Community guide)
+    * sonar.cs.dotcover.reportsPaths
+        * Path to dotCover coverage report.
+    * sonar.cs.opencover.reportsPaths
+        * Path to OpenCover coverage report.
+    * sonar.vbnet.dotcover.reportsPaths
+        * Path to dotCover coverage report.
+    * sonar.vbnet.opencover.reportsPaths
+        * Path to OpenCover coverage report.
+
+* Did you specify the correct path to your code coverage file?
+
+    Make sure you understand the regular expression for matching files and directories:
+    |Symbol|Meaning|
+    |---|---|
+    |?|a single character|
+    |*|any number of characters|
+    |**|any number of directories|
+    
+    Here is a resource to understand some other examples: [Pattern matching guide](https://confluence.atlassian.com/fisheye/pattern-matching-guide-960155410.html)
+
+    Examples of Correct Usage:
+    
+    Right :o:
+    
+    :o: `-d:"sonar.cs.vscoveragexml.reportsPaths=**/*.coveragexml”`
+    
+    :o: `/d:"sonar.cs.vscoveragexml.reportsPaths=coverage/joe*.coveragexml,otherdir/olivier*.coveragexml"`
+    
+    :o: `/d:"sonar.cs.vscoveragexml.reportsPaths=**/jojo_jt 2020-10-14 19_02_26.coveragexml"`(`-` and `/` are valid flag markers)
+    
+    :o: `-d:"sonar.cs.vscoveragexml.reportsPaths=coveragexml"` (assuming you are running SonarScanner.MSBuild.exe in same directory as your *.coveragexml file)
+    
+    :o: `-d:"sonar.cs.dotcover.reportsPaths=**/*.html"`
+    
+    :o: `-d:"sonar.vbnet.opencover.reportsPaths=**/someNameCodeCoverage.xml"`
+    
+    Wrong :x:
+    
+    :x:  Don't point at directories but files
+        `-d:"sonar.cs.vscoveragexml.reportsPaths=coverage/"`
+        
+    :x: Don't use absolute paths, favor paths relative to the project root directory
+        `-d:"sonar.cs.vscoveragexml.reportsPaths=C:\users\joe\project\coverage\*.xml"`
+        
+    :x: Don't point files that are not under the project root directory, always generate coverage under this home directry
+        `-d:"sonar.cs.vscoveragexml.reportsPaths=..\coverage\*.xml"`
+
+* Do you know how to get debug logs?
+
+    Add `/d:"sonar.verbose=true"` to the...
+    * SonarScanner.MSBuild.exe `begin` command to get more detailed logs (TODO======>should mention the alternative for Azure Devops integration, where the Scanner for MSBuild is hidden)
+    * `SonarCloudPrepare` task in case you are using Azure DevOps
+    
+        * For example: `SonarScanner.MSBuild.exe begin /k:"MyProject" /d:"sonar.verbose=true"`
```

Comment:
this should be between `SonarScanner.MSBuild.exe `begin`` and ``SonarCloudPrepare` task`